My Comments Feature

itbel opened the pull request:
* [x] Make "My Comments" clickable to a page with all of the comments the user has ever made
* [x] Clicking on a series graphic should take user to series page
* [x] Clicking on a comment should take the user to editing that comment where it exists (i.e. teaching - series - notes - comment editing)
* [x] Comments are searchable
* [x] Display comments by most recent
* [x] Display comments by series



┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=650425481)

itbel committed:
added MyComments screen
fixed ToggleButton component

itbel committed:
cleaned up ToggleButton, removed unused imports

itbel committed:
implemented comments flatlist

itbel committed:
conditional rendering based on ToggleButton placeholder
searchbar component added functionality missing
minor styling changes

itbel committed:
partially implemented rudimentary search

itbel committed:
added tag search

itbel committed:
loadComments logic WIP

itbel committed:
sorting by date

itbel committed:
sectionList partially implemented

itbel committed:
added series image on By Series view

itbel committed:
Added ability to click on comment to edit
Added ability to navigate to a series by clicking the series image

itbel committed:
removed any types

itbel committed:
ActivityIndicators for each list have been added

itbel committed:
fixed navigation to serieslandingscreen, fixed comment type

itbel committed:
improved performance for loading comments

itbel committed:
added load more

itbel committed:
updated comments

itbel committed:
now sorting by date(again)

itbel committed:
series image now uses FallbackImage component

itbel committed:
fixed iOS bugs, sticky header

itbel committed:
search now works for by series

itbel committed:
cleaned up code

itbel committed:
comment

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+/* eslint-disable no-await-in-loop */
```
ESLint disables are fine, but I would use `// eslint-disable-next-line` because it specifically marks the line(s) where you have elected to ignore the rule, which is helpful when others read your code.

JonEsparaz replied:
Also, if you disable it for the entire file, you might accidently ignore the rule when you didn't actually want to.

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-nested-ternary */
```
You use nested ternaries pretty often, so it seems this rule conflicts with how you use JSX.

Since you are the primary developer on this project now, perhaps disable this rule in the project's ESLint config?

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+              );
+            return 0;
+          });
+          await loadSeriesData(transformed);
```
It is a little unclear where the `setIsLoading(false)` call is here (since it lives in another function).

I would put a single `setIsLoading(false)` in a `finally` block.

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+
+      if (json.data?.getCommentsByOwner?.items) {
+        const commentData = json.data?.getCommentsByOwner?.items;
+        const transformed = commentData.map((comm) => {
```
Nit: a better variable name might be `commentWithSeriesInfo`.

Also, the variable `comm` could be `comment`, then you can use shorthand syntax: 

```typescript
return {
  comment,
  seriesInfo: {
    year: '',
    episodeNumber: 0,
    episodeTitle: '',
  },
} as Comment;
```

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+            return null;
+          }}
+          keyExtractor={(item, index) => {
+            return item.comment?.id ?? index.toString();
```
This would be an appropriate place to do `item.comment?.id as string` since the comment must have an `id` (otherwise it would not be in the database).

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,521 @@
+              </View>
+            ) : null
+          }
+          keyExtractor={(item, index) => {
```
`item.comment?.id as string` (same as above).

itbel commented:
Thanks for the feedback. I'll be pushing changes based on your input by the end of the day.

JonEsparaz commented:
Sounds good. Nice work here @itbel - should be a cool feature!

itbel committed:
disabled no nested ternary rule
no-await-in-loop disabled inline instead of entire file
keyExtractor for lists, now as string instead of toString()

itbel commented:
Thanks! I'm excited for people to try it.

itbel committed:
changed variable name, moved setIsLoading to finally

JonEsparaz commented on src/screens/comments/MyComments.tsx:
```diff
@@ -0,0 +1,523 @@
+  const [filterToggle, setFilterToggle] = useState(false);
+
+  const loadSeriesData = async (userComments: RecentComments) => {
+    setIsLoading(true);
```
I think you don't need the two `setIsLoading()` calls in this function. 

`loadComments` already does this and `loadSeriesData` is exclusively called within `loadComments`.

itbel replied:
Done. Thank you

itbel committed:
removed redundant setIsLoading from loadSeriesData

JonEsparaz commented:
One more thing, maybe I missed a design decision, but isn't the My Comments button supposed to be in ProfileScreen.tsx, not MoreScreen.tsx?

Directly from Figma:
<img src=https://user-images.githubusercontent.com/48423418/109699544-b2b0b700-7b5e-11eb-9354-3749242a4e84.png width=250>

<img src=https://user-images.githubusercontent.com/48423418/109699553-b47a7a80-7b5e-11eb-8f21-14106753e873.png width=250> 

@lucastbelem?

lucastbelem commented:
Hey team,

No design change - My Comments should appear in the profile page :)

On Tue, Mar 2, 2021 at 1:56 PM Jon Esparaz <notifications@github.com> wrote:

> One more thing, maybe I missed a design decision, but isn't the My
> Comments button supposed to be in ProfileScreen.tsx, not MoreScreen.tsx?
>
> Directly from Figma:
>
> <https://user-images.githubusercontent.com/48423418/109699544-b2b0b700-7b5e-11eb-9354-3749242a4e84.png>
>
>
> <https://user-images.githubusercontent.com/48423418/109699553-b47a7a80-7b5e-11eb-8f21-14106753e873.png>
>
> @lucastbelem <https://github.com/lucastbelem>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/themeetinghouse/mobile/pull/237#issuecomment-789135333>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALQO6JTCRNH3HRFI6ELAA53TBUYFZANCNFSM4YNDG24A>
> .
>


-- 

*Lucas Belem*

Communications Project Manager

The Meeting House
2700 Bristol Circle | Oakville, ON | L6H 6E1
p: 905.287.7000, ext. 821 | f: 905.287.7099

www.themeetinghouse.com



*This email, including attachments, is for the sole use of the intended
recipient and may contain confidential information.  If you are not the
intended recipient, any disclosure, copying or distribution of the
information is prohibited.  If this email was received in error, please
notify us by reply email and delete the original message.*

GeorgeBellTMH commented:
Just want to verify as well. my comments instead of my notes?

G

Sent from my iPhone

> On Mar 2, 2021, at 2:02 PM, Lucas Belem <notifications@github.com> wrote:
> 
> ﻿
> Hey team,
> 
> No design change - My Comments should appear in the profile page :)
> 
> On Tue, Mar 2, 2021 at 1:56 PM Jon Esparaz <notifications@github.com> wrote:
> 
> > One more thing, maybe I missed a design decision, but isn't the My
> > Comments button supposed to be in ProfileScreen.tsx, not MoreScreen.tsx?
> >
> > Directly from Figma:
> >
> > <https://user-images.githubusercontent.com/48423418/109699544-b2b0b700-7b5e-11eb-9354-3749242a4e84.png>
> >
> >
> > <https://user-images.githubusercontent.com/48423418/109699553-b47a7a80-7b5e-11eb-8f21-14106753e873.png>
> >
> > @lucastbelem <https://github.com/lucastbelem>?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/themeetinghouse/mobile/pull/237#issuecomment-789135333>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/ALQO6JTCRNH3HRFI6ELAA53TBUYFZANCNFSM4YNDG24A>
> > .
> >
> 
> 
> -- 
> 
> *Lucas Belem*
> 
> Communications Project Manager
> 
> The Meeting House
> 2700 Bristol Circle | Oakville, ON | L6H 6E1
> p: 905.287.7000, ext. 821 | f: 905.287.7099
> 
> www.themeetinghouse.com
> 
> 
> 
> *This email, including attachments, is for the sole use of the intended
> recipient and may contain confidential information. If you are not the
> intended recipient, any disclosure, copying or distribution of the
> information is prohibited. If this email was received in error, please
> notify us by reply email and delete the original message.*
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

itbel commented:
Nice catch @JonEsparaz! 
I'm on it

JonEsparaz commented:
> Just want to verify as well. my comments instead of my notes? G

This is the My Comments feature and there aren't any buttons that say "notes" on Profile screen or More screen.

@GeorgeBellTMH I'm not sure I follow your question - can you clarify?

itbel committed:
moved My Comments to Profile screen as per figma

GeorgeBellTMH commented:
So I think functionally people are taking notes on a Sunday morning...not commenting on the notes (comment implies they are public). I realize we have comment elsewhere in the UI, but I'm just not sure it makes sense. 

Lucas and David might have some thoughts on this, but my suggestion is that this should be renamed my notes, and the comment feature should become add a note...

JonEsparaz commented:
I see. Comment might be confusing, but using "note"/"notes" to describe two different things is confusing too IMO.

I can't think of a better word at the moment though.

JonEsparaz commented:
The YouVersion app says "Private Note" when you are adding notes on passages.

"Private Note" (differentiates from the teaching notes) or "Private Comment" (clearly states that your comments are not public) provides a good amount of clarity.

GeorgeBellTMH commented:
Lets talk about it at the meeting today with @deeinstereo ... just playing with it now, I think it might be also useful the first time you open a note to explain the note taking feature in a modal or bubble or something...it's not obvious from the UI that clicking on the text lets you comment...("click on the text to add your own notes" or something to that effect)
